drivers/gpio/gpio_rzt2m.c

```diff
@@ -27,26 +39,55 @@
 #define PULL_UP    (1 << PULL_SHIFT)
 #define PULL_DOWN  (2 << PULL_SHIFT)
 
+#define INT_INVERT       0
+#define INT_FALLING_EDGE 1
+#define INT_RISING_EDGE  2
+#define INT_BOTH_EDGE    3
+
+#define IRQ_COUNT    16
+#define NS_IRQ_COUNT 14
+
+#define MAX_PORT_SIZE 8
+
+#define RZT2M_GPIO_VALUE_IDENTITY(i, _) i
+
 struct rzt2m_gpio_config {
 	struct gpio_driver_config common;
+	uint8_t pin_irqs[MAX_PORT_SIZE];
 	uint8_t *port_nsr;
 	uint8_t *ptadr;
 	uint8_t port;
 };
 
 struct rzt2m_gpio_data {
 	struct gpio_driver_data common;
+	sys_slist_t cb;
+};
+
+struct rzt2m_gpio_common_config {};
+
+struct rzt2m_gpio_irq_slot {
+	const struct device *dev;
+	uint8_t pin;
+};
+
+struct rzt2m_gpio_common_data {
+	struct rzt2m_gpio_irq_slot irq_registered_ports[IRQ_COUNT];
 };
 
+static struct rzt2m_gpio_common_config rzt2m_gpio_common_config = {};
+
+static struct rzt2m_gpio_common_data rzt2m_gpio_common_data = {};
```

Comment:
No need for initializers here?

Reply:
They are not needed and I've removed them. At the same time I had to rename those because the same name as the type name without initializer was causing ci compliance errors